File: server-scala-play-server/app/model/Shipment.scala
package model

import play.api.libs.json._
import java.time.OffsetDateTime

/**
  * Represents the Swagger definition for Shipment.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2026-02-09T20:57:39.376804970Z[Etc/UTC]", comments = "Generator version: 7.20.0-SNAPSHOT")
case class Shipment(
  id: Option[Long],
  active: Option[Boolean],
  valid: Option[Boolean],
  name: Option[String],
  owner: Option[Account],
  rider: Option[Account],
  pickupStop: Option[Stop],
  dropOffStop: Option[Stop],
  notes: Option[List[Note]],
  noteCount: Option[Long],
  batchId: Option[String],
  reserved: Option[OffsetDateTime],
  canceled: Option[OffsetDateTime],
  confirmed: Option[OffsetDateTime],
  route: Option[Route],
  polyline: Option[String],
  replacePassenger: Option[Shipment],
  refunded: Option[Boolean],
  promoCode: Option[PromoCode],
  quantity: Option[Int],
  pickupTimeWindow: Option[DateTimeRange],
  dropOffTimeWindow: Option[DateTimeRange],
  cargoType: Option[CargoType],
  serviceType: Option[ServiceType],
  routeId: Option[Long],
  totalVolume: Option[Double],
  complete: Option[Boolean],
  contentName: Option[String],
  riderName: Option[String]
)

object Shipment {
  implicit lazy val shipmentJsonFormat: Format[Shipment] = Json.format[Shipment]
}

File: server-aspnet-fastendpoints/src/Org.OpenAPITools/Models/Shipment.cs
namespace Org.OpenAPITools.Models;


/// <summary>
/// 
/// </summary>
public class Shipment 
{
    public long Id { get; set; }
    public bool Active { get; set; }
    public bool Valid { get; set; }
    public string Name { get; set; }
    public Account Owner { get; set; }
    public Account Rider { get; set; }
    public Stop PickupStop { get; set; }
    public Stop DropOffStop { get; set; }
    public List<Note> Notes { get; set; }
    public long NoteCount { get; set; }
    public string BatchId { get; set; }
    public DateTime Reserved { get; set; }
    public DateTime Canceled { get; set; }
    public DateTime Confirmed { get; set; }
    public Route Route { get; set; }
    public string Polyline { get; set; }
    public Shipment ReplacePassenger { get; set; }
    public bool Refunded { get; set; }
    public PromoCode PromoCode { get; set; }
    public int Quantity { get; set; }
    public DateTimeRange PickupTimeWindow { get; set; }
    public DateTimeRange DropOffTimeWindow { get; set; }
    public CargoType CargoType { get; set; }
    public ServiceType ServiceType { get; set; }
    public long RouteId { get; set; }
    public double TotalVolume { get; set; }
    public bool Complete { get; set; }
    public string ContentName { get; set; }
    public string RiderName { get; set; }
}


